src/utility.rs

```diff
@@ -63,8 +63,10 @@ pub fn compute_average(slice: &[f32], weights: &[f32]) -> f32 {
     let mut weight_sum = 0.0;
     let mut value_sum = 0.0;
     for (&v, &w) in slice.iter().zip(weights.iter()) {
-        weight_sum += w as f64;
-        value_sum += v as f64 * w as f64;
+        if w != 0.0 {
```

Comment:
@bkushigian Changed this so that `compute_average` will still work in cases where values are `NaN` so long as their weights are `0`.

Reply:
Hmmm, I'm not sure how much this is used but branching is not good for performance. If this is used in central solver logic then I think we can't do this.

Reply:
I can't look right now. But I'd rather have a way to either replace nans with 0.0 or have a specialized function that handles this differently.

Reply:
Will change this to a caller-side check.

Reply:
Now using `compute_average_filter_0s` instead